SignalExtraction/fitWm.C

```diff
@@ -744,167 +807,245 @@ void fitWm(const TString  outputDir,   // output directory
+              if(pileupDown)  recoilCorrPuDown->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0);
+              if(doKeys){
+                if(fabs(genVy)<0.5)
+                  recoilCorrKeys05->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys);
+                else if (fabs(genVy)>=0.5 && fabs(genVy)<1.0)
+                  recoilCorrKeys051->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys);
+                else
+                  recoilCorrKeys1->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys); 
+              } else {
+                if(fabs(genVy)<0.5)
+                  recoilCorr05->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys);
+                else if (fabs(genVy)>=0.5 && fabs(genVy)<1.0)
+                  recoilCorr051->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys);
+                else
+                  recoilCorr1->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys); 
+              }
+              // Compute the corrected MET value
+              TVector2 vMetCorr((corrMet)*cos(corrMetPhi),(corrMet)*sin(corrMetPhi));
+              Double_t corrMetWithLepton = (vMetCorr + vLepRaw - vLepCor).Mod();
+              if(typev[ifile]==eWmunu)
+              {
+                hWmunuMetp->Fill(corrMetWithLepton,weight);
+//                 hWmunuMetp_PileupUp->Fill(corrMetWithLepton,weightUp);
+//                 hWmunuMetp_PileupDown->Fill(corrMetWithLepton,weightDown);
+              }
+              else
+              {
+                hEWKMetp->Fill(corrMetWithLepton,weight);
+              }
+
               corrMet=met, corrMetPhi=metPhi;
             } else {
-              pU1 = 0; pU2 = 0; 
-              hWmunuMetm_RecoilDown->Fill(corrMet,weight);
+              if(doInclusive) recoilCorrm->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0);
+              if(pileupUp)    recoilCorrPuUpm->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0);
+              if(pileupDown)  recoilCorrPuDownm->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0);
+              if(doKeys){
+                if(fabs(genVy)<0.5)
+                  recoilCorrKeysm05->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys);
+                else if (fabs(genVy)>=0.5 && fabs(genVy)<1.0)
+                  recoilCorrKeysm051->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys);
+                else
+                  recoilCorrKeysm1->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys); 
+              } else {
```

Comment:
ok I decided to make it into if-else statements since there are several options and they are all exclusive of each other